wfe/web-front-end.go

```diff
@@ -478,11 +484,20 @@ func (wfe *WebFrontEndImpl) Challenge(authz core.Authorization, response http.Re
 			}
 			return
 		}
-		if currReg.Agreement != wfe.SubscriberAgreementURL {
+		// Any version of the agreement is acceptable here. Version match is enforced in
+		// UpdateRegistration when agreeing the first time. Agreement updates happen
+		// by mailing subscribers and don't require a registration update.
+		if currReg.Agreement == "" {
 			wfe.sendError(response, "Must agree to subscriber agreement before any further actions", nil, http.StatusUnauthorized)
 			return
 		}
 
+		var challengeResponse core.Challenge
+		if err = json.Unmarshal(body, &challengeResponse); err != nil {
+			wfe.sendError(response, "Error unmarshaling challenge response", err, http.StatusBadRequest)
+			return
+		}
+
```

Comment:
Is there a reason you're moving this stuff back up?  It seemed kind of prudent to me to check that the registration ID was right before we attempt to parse.  In any case, this change seems unrelated to the topic of this PR.